OpenRA.Game/Traits/Player/PlayerResources.cs

```diff
@@ -85,6 +85,19 @@ public PlayerResources(Actor self, PlayerResourcesInfo info)
 
 		int lastNotificationTick;
 
+		public int ChangeCash(int amount)
+		{
+			if (amount >= 0)
+				GiveCash(amount);
+			else
+			{
+				amount = Math.Min(Cash + Resources, -amount);
+				TakeCash(-amount);

```

Comment:
Are you sure this is supposed to be `-amount`?

Reply:
It seems like this is causing negative cash to go to the player's resources instead!

Reply:
With all the rebasing I'm not sure when was the last time I touched this. It feels that I would code something like this.

Anyway, are there any other issues I should know about? I know I keep asking this, but every time I do you tell me it's ok only for more issues to pop-up.

Reply:
I took a look and as unlikely as it may seem, it is supposed to be `-amount`.
The take cash method operates with positive numbers. I could give it a rewrite, but honestly it makes logical sense for it to behave as it does now.
`TakeCash(howMuchCashIWantToTake)`

I will take a look at the first issue you raised later.

Reply:
Well turns out I was wrong, again. The Min method already flips the sign on this one so it should have been `TakeCash(amount)`